### PR TITLE
refactor: Removed test target, added convenience docker-all target & removed ...

### LINKED ISSUE #1489: make targets missing or fail
## Summary of Bug

`make all` fails with ... make: *** No rule to make target `test', needed by `all'.  Stop.
`make start-localnet-ci` fails with ... unknown flag: --mode make: *** [start-localnet-ci] Error 1

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by mmulji-ic**
… unused target for contracts

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -221,7 +223,7 @@ format:
 ###                                Localnet                                 ###
 ###############################################################################
 
-start-localnet-ci:
+start-localnet-ci: build
 	./build/gaiad init liveness --chain-id liveness --home ~/.gaiad-liveness
```
Comment (mmulji-ic) [inline Makefile:224]: Would like to add a `rm -rf here ~/.gaiad-liveness`, opinions?

Comment (yaruwangway) [reply Makefile:224]: maybe not ? in case people still want to keep the data.  let them manually `rm`

Comment (okwme) [reply Makefile:224]: liveness test shouldn't ever be persisted though. it's just to make sure that blocks are produced. i'm ok with removing as long as we're sure there is no chance liveness tests are being configured with custom home directories or anything like that, right?

Comment (mmulji-ic) [reply Makefile:224]: We do use a custom home, but this target is really just for the CI & occasional local testing

Comment (okwme) [reply Makefile:224]: ok let's add the `rm`

File: Makefile [hunk 2]
```diff
@@ -249,4 +248,4 @@ test-docker-push: test-docker
 	@docker push ${TEST_DOCKER_REPO}:latest
```
Comment (mmulji-ic) [inline Makefile:248]: @okwme do we still want this two docker commands that rely on contrib/Dockerfile.test?
The docker push, pushes 3 things, latest, branch name & commit hash. Think branch name/latest are ok.

Comment (okwme) [reply Makefile:248]: i'm ok with keeping as is if there's no large cost, in time or money, in doing so

**Review by okwme — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -221,15 +219,16 @@ format:
...
+	rm -rf ~/.gaiad-liveness
 	./build/gaiad init liveness --chain-id liveness --home ~/.gaiad-liveness
 	./build/gaiad config chain-id liveness --home ~/.gaiad-liveness
 	./build/gaiad config keyring-backend test --home ~/.gaiad-liveness
 	./build/gaiad keys add val --home ~/.gaiad-liveness
 	./build/gaiad add-genesis-account val 10000000000000000000000000stake --home ~/.gaiad-liveness --keyring-backend test
 	./build/gaiad gentx val 1000000000stake --home ~/.gaiad-liveness --chain-id liveness
 	./build/gaiad collect-gentxs --home ~/.gaiad-liveness
-	sed -i'' 's/minimum-gas-prices = ""/minimum-gas-prices = "0uatom"/' ~/.gaiad-liveness/config/app.toml
+	sed -i.bak'' 's/minimum-gas-prices = ""/minimum-gas-prices = "0uatom"/' ~/.gaiad-liveness/config/app.toml
```
Comment (mmulji-ic) [inline Makefile:231]: Works on Mac, if theres a bak file created.

**Pull request merged**